airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py

```diff
@@ -64,64 +64,91 @@ class UnitTests(GradleTask):
-            step_results.append(build_normalization_results)
         else:
             normalization_tar_file = None
 
-        connector_container = build_connector_image_results.output_artifact[LOCAL_BUILD_PLATFORM]
-        connector_image_tar_file, _ = await export_container_to_tarball(context, connector_container)
 
-        async with asyncer.create_task_group() as docker_build_dependent_group:
-            soon_integration_tests_results = docker_build_dependent_group.soonify(IntegrationTests(context).run)(
-                connector_tar_file=connector_image_tar_file, normalization_tar_file=normalization_tar_file
-            )
-            soon_cat_results = docker_build_dependent_group.soonify(AcceptanceTests(context, True).run)(connector_container)
+        return {
+            "connector_tar_file": connector_image_tar_file,
+            "normalization_tar_file": normalization_tar_file
+        }
+
+    return _create_integration_step_args
+
+def _get_normalization_steps(context: ConnectorContext) -> List[StepToRun]:
+    normalization_image = f"{context.connector.normalization_repository}:dev"
+    context.logger.info(f"This connector supports normalization: will build {normalization_image}.")
+    normalization_steps = [
+        StepToRun(
+            id="build_normalization",
+            step=BuildOrPullNormalization(context, normalization_image, LOCAL_BUILD_PLATFORM),
+            depends_on=["build"],
+        )
+    ]
 
-        step_results += [soon_cat_results.value, soon_integration_tests_results.value]
-        return step_results
+    return normalization_steps
 
-    async with asyncer.create_task_group() as test_task_group:
-        soon_unit_tests_result = test_task_group.soonify(UnitTests(context).run)()
-        soon_docker_build_dependent_steps_results = test_task_group.soonify(run_docker_build_dependent_steps)(dist_tar_dir)
+def _get_acceptance_test_steps(context: ConnectorContext) -> List[StepToRun]:
+    """
+    Generate the steps to run the acceptance tests for a Java connector.
+    """
+    # Run tests in parallel
+    return [
+        StepToRun(
+            id="integration",
+            step=IntegrationTests(context),
+            args=_create_integration_step_args_factory(context),
+            depends_on=["build"],
+        ),
+        StepToRun(
+            id="acceptance",
+            step=AcceptanceTests(context, True),
+            args=lambda results: {"connector_under_test_container": results["build"].output_artifact[LOCAL_BUILD_PLATFORM]},
+            depends_on=["build"],
+        ),
+    ]
+
+
+def get_test_steps(context: ConnectorContext) -> List[StepToRun]:
+    """
+    Get all the tests steps for a Java connector.
+    """
 
-    return step_results + [soon_unit_tests_result.value] + soon_docker_build_dependent_steps_results.value
+    steps = [
+        [StepToRun(id="build_tar", step=BuildConnectorDistributionTar(context))],
+        [StepToRun(id="unit", step=UnitTests(context))],
```

Comment:
```suggestion
        [StepToRun(id="build_tar", step=BuildConnectorDistributionTar(context))],
        [StepToRun(id="unit", step=UnitTests(context), depends_on=["build_tar"])],
```
Even if there are no functional requirement to make unit test depend on build tar I'd prefer to express a dependency here as `build_tar` will warm the unit test gradle cache.